be/src/olap/comparison_predicate.h

```diff
@@ -107,8 +120,22 @@ class ComparisonPredicateBase : public ColumnPredicate {
         }
 
         roaring::Roaring roaring;
-        RETURN_IF_ERROR(iterator->read_from_inverted_index(column_name, &_value, query_type,
-                                                           num_rows, &roaring));
+        if constexpr (Type == PrimitiveType::TYPE_DATE) {
+            uint24_t date = _value.to_olap_date();
```

Comment:
Maybe we can uniformly implement a function for converting CppType and StorageType to each other.